### PR TITLE
Implement custom video loader caching, disable Webpack caching

### PR DESCRIPTION
**Pull request by MattIPv4**
A solution to the continued Webpack cache issues investigated in #268

https://github.com/alveusgg/alveusgg/pull/269/commits/0d3ab4ab5ad53a6465ec74d6cc16e0389658bba6 (no existing cache): https://vercel.com/alveus/alveusgg-website/BAijGm2Wv9N9Taw64HCvz3SG1soE 366.29 MB cache in ~7m

https://github.com/alveusgg/alveusgg/pull/269/commits/37bde3a5843e31e561c18d9f9f756c7eb68f5e45 (cache from above): https://vercel.com/alveus/alveusgg-website/3WDhweMUvEYMhoE7iuTqqrTppjtB 366.29 MB cache in ~3m

### CODE REVIEW

**Comment by MattIPv4** [comment]
@pjeweb what do you think of this as a solution?

File: apps/website/build-scripts/video-loader.cjs [hunk 1]
```diff
@@ -266,26 +343,77 @@ const videoLoader = async (context, content) => {
...
+  };
 };
 
+/**
+ * Webpack hook to clean up old cache files
+ *
+ * @return {Promise<void>}
+ */
+const cacheCleanup = async () => {
+  // Get all files in the cache dir
```
Comment (pjeweb) [inline apps/website/build-scripts/video-loader.cjs:374]: this should handle the case if the cacheDir does not exist yet. that might happen when doing a clean `pnpm build`.
Something like `if (!fs.existsSync(cacheDir)) return;`

**Review by pjeweb — DISMISSED** [review]
The solution makes sense to me. I would not want to debug webpack any further. So unless you want to, I am happy with this.

File: apps/website/build-scripts/video-loader.cjs (hunk 1 above)
Comment (MattIPv4) [reply apps/website/build-scripts/video-loader.cjs:374]: 🤔 It should never be possible for this directory to not exist -- the loader always runs before this, and always writes to the cache dir?

Comment (MattIPv4) [reply apps/website/build-scripts/video-loader.cjs:374]: I guess I should handle the case that something manages to remove that directory between the loader running and the cleanup plugin running 👍

Comment (pjeweb) [reply apps/website/build-scripts/video-loader.cjs:374]: It happened to me. I 'rm -rf'ed .next and ran 'pnpm build'

**Review by pjeweb — APPROVED** [review]

File: apps/website/build-scripts/video-loader.cjs (hunk 1 above)
Comment (MattIPv4) [reply apps/website/build-scripts/video-loader.cjs:374]: Oh right yes, the server build happens first that doesn't emit assets 🤦

**Pull request merged**